data/scripts/globalevents/server_initialization.lua

```diff
@@ -0,0 +1,155 @@
+				local highestBidder = Result.getNumber(resultId, "highest_bidder")
+				local balance = Result.getNumber(resultId, "balance")
+				local lastBid = Result.getNumber(resultId, "last_bid")
+				if balance >= lastBid then
+					db.query("UPDATE `players` SET `balance` = " .. (balance - lastBid) .. " WHERE `id` = " .. highestBidder)
+					house:setOwnerGuid(highestBidder)
+				end
+
+				db.asyncQuery("UPDATE `houses` SET `last_bid` = 0, `bid_end` = 0, `highest_bidder` = 0, `bid` = 0 " .. "WHERE `id` = " .. house:getId())
+			end
+		until not Result.next(resultId)
+
+		Result.free(resultId)
+	end
+end
+
+-- Function to store towns in the database
+local function storeTownsInDatabase()
+	db.query("TRUNCATE TABLE `towns`")
+
+	for i, town in ipairs(Game.getTowns()) do
+		local position = town:getTemplePosition()
+		db.query("INSERT INTO `towns` (`id`, `name`, `posx`, `posy`, `posz`) VALUES (" .. town:getId() .. ", " .. db.escapeString(town:getName()) .. ", " .. position.x .. ", " .. position.y .. ", " .. position.z .. ")")
+	end
+end
+
+-- Function to check for duplicate keys in a given variable's storage
+local function checkDuplicateStorageKeys(varName)
+	local keys = _G[varName]
+	local seen = {}
+	local duplicates = {}
+
+	for k, v in pairs(keys) do
+		if seen[v] then
+			table.insert(duplicates, v)
+		else
+			seen[v] = true
+		end
+	end
+
+	return next(duplicates) and duplicates or false
+end
+
+-- Function to check duplicated variable keys and log the results
+local function checkAndLogDuplicateKeys(variableNames)
+	for _, variableName in ipairs(variableNames) do
+		local duplicates = checkDuplicateStorageKeys(variableName)
+		if duplicates then
+			local message = "Duplicate keys found: " .. table.concat(duplicates, ", ")
+			logger.warn("Checking " .. variableName .. ": " .. message)
+		else
+			logger.info("Checking " .. variableName .. ": No duplicate keys found.")
+		end
+	end
+end
+
+-- Function to update event rates based on EventsScheduler data
+local function updateEventRates()
+	local lootRate = EventsScheduler.getEventSLoot()
+	if lootRate ~= 100 then
+		SCHEDULE_LOOT_RATE = lootRate
+	end
+
+	local expRate = EventsScheduler.getEventSExp()
+	if expRate ~= 100 then
+		SCHEDULE_EXP_RATE = expRate
+	end
+
+	local skillRate = EventsScheduler.getEventSSkill()
+	if skillRate ~= 100 then
+		SCHEDULE_SKILL_RATE = skillRate
+	end
+
+	local spawnRate = EventsScheduler.getSpawnMonsterSchedule()
+	if spawnRate ~= 100 then
+		SCHEDULE_SPAWN_RATE = spawnRate
+	end
+
+	-- Log information if any of the rates are not 100%
+	if expRate ~= 100 or lootRate ~= 100 or spawnRate ~= 100 or skillRate ~= 100 then
+		logger.info("[Events] Exp: {}%, loot: {}%, Spawn: {}%, Skill: {}%", expRate, lootRate, spawnRate, skillRate)
+	end
+end
+
+-- Function to reset account sessions based on configuration and authentication type
+local function resetAccountSessions()
+	local resetSessionsOnStartup = configManager.getBoolean(configKeys.RESET_SESSIONS_ON_STARTUP)
```

Comment:
local resetSessionsOnStartup: não precisa criar uma váriavel onde o conteúdo dela será utilizado apenas uma vez no código. (e ainda se o AUTH_TYPE == "session" for falso, ela nem será usada)